sdk/python/kfp/v2/compiler/compiler.py

```diff
@@ -0,0 +1,308 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""KFP DSL v2 compiler implementation."""
+
+import contextlib
+import inspect
+import io
+import tarfile
+from typing import Any, Callable, List, Optional
+import zipfile
+
+import kfp
+from kfp import dsl
+from kfp.compiler._k8s_helper import sanitize_k8s_name
+from kfp.components.structures import ComponentSpec, InputSpec
+from kfp.dsl import ir_types
+from kfp.dsl._metadata import _extract_pipeline_metadata
+from kfp.v2.compiler import importer_node
+from kfp.v2.proto import pipeline_spec_pb2
+
+from google.protobuf.json_format import MessageToJson
+
+
+class Compiler(object):
+  """DSL Compiler that compiles pipeline functions into pipeline_spec json.
```

Comment:
What is pipeline_spec json? Elaborate a little here and point to IR proto if possible.

Reply:
I think you should also call out limitations here, i.e., control flow etc are not fully supported yet.

Reply:
Done